app/api.py

```diff
@@ -438,14 +439,19 @@ def get(self):
                 results = [person]
         elif query_string:
             # Search by query words.
-            query = TextQuery(query_string)
             if self.config.external_search_backends:
+                query = TextQuery(query_string)
                 results = external_search.search(self.repo, query, max_results,
                     self.config.external_search_backends)
             # External search backends are not always complete. Fall back to
             # the original search when they fail or return no results.
             if not results:
-                results = indexing.search(self.repo, query, max_results)
+                if self.config.enable_fulltext_search:
```

Comment:
Can you use config.get('enable_fulltext_search') instead? self.config.enable_fulltext_search supports repository-specific config but config.get('enable_fulltext_search') doesn't. So mixing these two can cause confusing behavior.